src/noit_metric.c

```diff
@@ -349,16 +349,31 @@ noit_metric_tagset_encode_tag_ex(char *encoded_tag, size_t max_len,
                                  noit_metric_encode_type_t left,
                                  noit_metric_encode_type_t right)
 {
-  char scratch[NOIT_TAG_MAX_PAIR_LEN+1];
-  if(max_len > sizeof(scratch)) return -1;
+  mtev_dyn_buffer_t scratch;
+  mtev_dyn_buffer_init(&scratch);
+  mtev_dyn_buffer_ensure(&scratch, max_len);
+  if(max_len > mtev_dyn_buffer_size(&scratch)) {
+    //TODO: I am keeping this check only to confirm 'mtev_dyn_buffer_ensure' succeeded (since it
```

Comment:
It's fine, don't need this

Reply:
we don't need the check or we don't need the comment?

Reply:
both